cql3/statements/alter_table_statement.cc

```diff
@@ -398,22 +401,21 @@ std::pair<schema_builder, std::vector<view_ptr>> alter_table_statement::prepare_
                         view->view_info()->where_clause(),
                         view_renames,
                         cql3::dialect{});
-                builder.with_view_info(view->view_info()->base_id(), view->view_info()->base_name(),
-                        view->view_info()->include_all_columns(), std::move(new_where));
+                builder.with_view_info(new_base_schema, view->view_info()->include_all_columns(), std::move(new_where));
                 view_updates.push_back(view_ptr(builder.build()));
```

Comment:
Shouldn't we unconditionally update the view schema? So it can refer to the new base schema, even if it's not affected directly?

Reply:
After this series, base_info only contains fields that do not change even if the base schema changes, so there's no need to update it unconditionally. 
Also, at the point of this specific commit, we were updating the base_info on each base schema change anyway, so it was also not useful to update the view schema unconditionally (note that here we prepare the schema mutations for the view schemas and the base_info was not included in schema mutations, it was set separately after applying the mutations)